module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -171,16 +171,27 @@ protected function getSearchServiceWithMockSearchMethod(
         array $expectedParams
     ): SearchService {
         $collection = new RecordCollection($response);
-        $searchService = $this->createMock(SearchService::class);
-        $searchService->expects($this->once())
-            ->method('search')
-            ->with(
-                $this->equalTo('Solr'),
-                $this->equalTo(new \VuFindSearch\Query\Query()),
-                $this->equalTo(0),
-                $this->equalTo(20),
-                $this->equalTo(new \VuFindSearch\ParamBag($expectedParams))
-            )->will($this->returnValue($collection));
+        $searchService = $this->getMockBuilder(\VuFindSearch\Service::class)
+        ->disableOriginalConstructor()
+        ->getMock();
+
+        $commandObj = $this->getMockBuilder(\VuFindSearch\Command\AbstractBase::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $commandObj->expects($this->once())->method('getResult')
+            ->will($this->returnValue($collection));
+
+        $checkCommand = function ($command) {
+            return get_class($command) === \VuFindSearch\Command\SearchCommand::class
+                && $command->getTargetIdentifier() === 'Solr'
+                && get_class($command->getArguments()[0]) === \VuFindSearch\Query\Query::class
+                && $command->getArguments()[1] === 0
+                && $command->getArguments()[2] === 20
+                && get_class($command->getArguments()[3]) == \VuFindSearch\ParamBag::class;
```

Comment:
Would it be more meaningful to compare the array copy of argument 3 against `$expectedParams` rather than only checking the class?